### ISSUE TITLE
How to fetch all bookmarks favicons?

### ISSUE THREAD

**Issue by Teko101** [aaFn/Bookmark-search-plus-2#215]
Hi,
I know it might be silly to ask that question, but I couldn't find the answer in the 'wiki' page. 
I've a huge bookmarks/favicons database and for some reason the 'favicons.sqlite' file is now corrupted.

Now, in the option page, I set "favicon Fetching" option to 'Active', and I noticed that the 'Statistics (number of bookmarks)' changes when I reload the option. So, is that it? or am I missing something here?

Best Regards :)

**Comment by aaFn** [comment]
Hello @Teko101 , the favicon.sqlite DB is the Firefox one to which add-ons (and so BSP2) cannot have access.
Therefore, BSP2 has to make its own favicon DB for displaying the favicons, and this is stored in local storage, a different place in Firefox.

When you enable favicon fetching in BSP2, it starts to fetch them in the background at a low rate to avoid impacting FF performances, and the counters in the options page show the progress and a remaining % to fetch. When this number is at 0.0 %, then no more favicons to fetch for BSP2.

The only safe way I know of for FF to reconstitute its own favicon DB is by accessing the bookmark pages one by one ... a long and manual process, so this is why I added the background fetch function in BSP2.
I cannot tell if at the same time that BSP2 fetches them, Firefox takes the opportunity of the BSP2 to also reconstitute its own favicon DB .. that is possible, but I wouldn't bet on it.

**Comment by Gitoffthelawn** [comment]
@Teko101 I have experienced Firefox's `favicons.sqlite` having issues as well.  What I do is disable JavaScript and load every site in each bookmarks folder in batches.  If the folder is small enough, I'll load the entire folder at once.  If the folder is too large, I'll select 100 or so bookmarks and open them all up.

It's a PITA, but I do it in batches while waiting on hold for some company that doesn't value my time.  Oh, and you can paint your toenails at the same time!

Firefox still has some issues with Favicons, but not nearly like it was years ago.  I hope they continue to work on those issues.

**Comment by aaFn** [comment]
Closing as the question is answered.
Thanks, aaFn.

**Comment by Gitoffthelawn** [comment]
> I cannot tell if at the same time that BSP2 fetches them, Firefox takes the opportunity of the BSP2 background page and favicon accesses to also reconstitute its own favicon DB .. that is possible, but I wouldn't bet on it.

I was just thinking (uh oh)... if you wanted to intentionally populate Firefox's favicon database at the same time as BSP2's favicon database, it *might* be possible to do so.  It would definitely be a nice feature.
